core/src/dpdk/mod.rs

```diff
@@ -9,6 +9,84 @@ include!(concat!(env!("OUT_DIR"), "/dpdk.rs"));
 
 use std::os::raw::{c_char, c_int, c_uint, c_void};
 
+#[cfg(dpdk_ge_2311)]
+impl std::fmt::Debug for rte_gtp_psc_generic_hdr {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("rte_gtp_psc_generic_hdr")
+            .field("ext_hdr_len", &self.ext_hdr_len)
+            .field("_bitfield_align_1", &self._bitfield_align_1)
+            .field("_bitfield_1", &self._bitfield_1)
+            .field("data", &self.data)
+            .finish()
+    }
+}
+
+impl rte_ipv4_hdr {
+    #[cfg(not(dpdk_ge_2311))]
+    pub fn set_version_ihl(&mut self, version_ihl: u8) {
+        self.version_ihl = version_ihl;
+    }
+
+    #[cfg(dpdk_ge_2311)]
+    pub fn set_version_ihl(&mut self, version_ihl: u8) {
+        self.__bindgen_anon_1.version_ihl = version_ihl;
+    }
+}
+
+// Bindgen doesn't support complex macros, so we need these
```

Comment:
I don't think this is true. See 

https://github.com/stanford-esrg/retina/blob/6ea7f1dfab8087fa29954765c8ca3a30674be5be/core/src/port/mod.rs#L298

Reply:
The problem is that in DPDK 23.11 these constants are defined with a function macro `RTE_BIT64(x)` and bindgen doesn't support function macros. In old DPDK these were ok.

Reply:
Got it - sorry for missing that. Ideally the code wouldn't redefine the constants from scratch so that we can better ensure consistency with the original DPDK. Though I think this could be put into an issue to be addressed in a later commit.

I did a quick search through GitHub and couldn't find other projects that have needed to do this with recent DPDK versions. 

I'm wondering if there's a clean way to force preprocessing to expand the function macros 
- Maybe there's something in build.rs: https://gcc.gnu.org/onlinedocs/gcc/Preprocessor-Options.html 
- Instead of these defs in a .rs file, I assume putting them in a C file would allow defining these consts with direct reference to the DPDK function macros. (E.g., `define RTE_ETH_RSS_IPV4_ RTE_ETH_RSS_IPV4` --> bindgen can expose these to the rest of the crate.) 

What do you think?

Reply:
I found out that new bindgen has a `clang_macro_fallback` which solves this problem. So I updated the bindgen and removed those manual copies.